### PR TITLE
Move config db testing methods to utility class

### LINKED ISSUE #8229: Refactor DbResourceGroupsConfigurationManager unit tests
Refactor the DbResourceGroupsConfigurationManager unit tests to be like other plugins.
This will make it easier to add more unit tests, including those in upcoming pr's.

Move testing utility methods to a single place so they can be reused.

### PR DESCRIPTION
**Pull request by elonazoulay**
Move test utility classes for DbResourceGroupConfigurationManager to H2TestingUtil.
Resolves #8229

### CODE REVIEW

**Comment by elonazoulay** [comment]
Thanks @raghavsethi ! Renamed, let me know if this is more descriptive.

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import static com.facebook.presto.execution.QueryState.RUNNING;
+import static com.facebook.presto.execution.QueryState.TERMINAL_QUERY_STATES;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class H2TestingUtil
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: H2TestUtil

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+{
+    static final String NAME = "h2";
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: Not descriptive enough. Maybe `RESOURCE_GROUP_MANAGER`? Also why is this not private?

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    private H2TestingUtil() {}
+
+    static Session newSession()
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: `adhocSession`

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    {
+        return testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema("sf100000")
+                .setSource("adhoc")
+                .build();
+    }
+
+    static Session newDashboardSession()
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: We generally use class visibility to control method visibility. Make the class package-private, and make the methods that need to be accessed outside this class public.

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    {
+        return testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema("sf100000")
+                .setSource("dashboard")
+                .build();
+    }
+
+    static Session newRejectionSession()
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: `rejectingSession`

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java (hunk 4 above)
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: `dashboardSession`

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    {
+        QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
+        while (queryManager.getAllQueryInfo().stream().filter(q -> countingStates.contains(q.getState())).count() != expectedCount) {
+            MILLISECONDS.sleep(500);
+        }
+    }
+
+    static String getDbConfigUrl()
+    {
+        Random rnd = new Random();
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: We never use abbreviations, fix in separate commit

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    static DistributedQueryRunner getSimpleQueryRunner()
+            throws Exception
+    {
+        String dbConfigUrl = getDbConfigUrl();
+        H2ResourceGroupsDao dao = getDao(dbConfigUrl);
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        builder.put("experimental.resource-groups-enabled", "true");
+        Map<String, String> properties = builder.build();
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: Same comment

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+            setup(queryRunner, dao);
+            return queryRunner;
+        }
+        catch (Exception e) {
+            queryRunner.close();
+            throw e;
+        }
+    }
+
+    static DistributedQueryRunner getSimpleQueryRunner()
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java]: Could you explain how these two methods are different? Can common logic be factored out?

**Comment by elonazoulay** [comment]
Hey @raghavsethi - does everything look ok with this pr?

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestUtil.java [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+            throws InterruptedException
+    {
+        waitForQueryCount(queryRunner, ImmutableSet.of(RUNNING), expectedCount);
+    }
+
+    static void waitForQueryCount(DistributedQueryRunner queryRunner, Set<QueryState> countingStates, int expectedCount)
+            throws InterruptedException
+    {
+        QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
+        while (queryManager.getAllQueryInfo().stream().filter(q -> countingStates.contains(q.getState())).count() != expectedCount) {
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestUtil.java]: Split into multiple lines

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestUtil.java [hunk 10]
```diff
@@ -115,7 +114,7 @@ public static DistributedQueryRunner createQueryRunner(String dbConfigUrl, H2Res
         ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
         builder.put("experimental.resource-groups-enabled", "true");
         Map<String, String> properties = builder.build();
-        DistributedQueryRunner queryRunner = new DistributedQueryRunner(testSessionBuilder().build(), 2, ImmutableMap.of(), properties, new SqlParserOptions());
+        DistributedQueryRunner queryRunner = new DistributedQueryRunner(testSessionBuilder().setCatalog("tpch").setSchema("tiny").build(), 2, ImmutableMap.of(), properties, new SqlParserOptions());
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestUtil.java]: Split stacked function calls onto separate lines

**Comment by elonazoulay** [comment]
Done. Thanks!

**Comment by raghavsethi** [comment]
Merged, thanks for your contribution!

**Pull request closed without merging**